crates/sui-framework/src/natives/mod.rs

```diff
@@ -9,100 +9,134 @@ mod transfer;
 mod tx_context;
 mod types;
 
-use move_binary_format::errors::PartialVMError;
+use crate::make_native;
+use move_binary_format::errors::{PartialVMError, PartialVMResult};
 use move_core_types::{account_address::AccountAddress, identifier::Identifier};
+use move_stdlib::natives::GasParameters;
 use move_vm_runtime::native_functions::{NativeFunction, NativeFunctionTable};
-use move_vm_types::values::{Struct, Value};
+use move_vm_types::{
+    natives::function::NativeResult,
+    values::{Struct, Value},
+};
+use std::sync::Arc;
 
 pub fn all_natives(
     move_stdlib_addr: AccountAddress,
     sui_framework_addr: AccountAddress,
 ) -> NativeFunctionTable {
-    const SUI_NATIVES: &[(&str, &str, NativeFunction)] = &[
-        ("crypto", "ecrecover", crypto::ecrecover),
-        ("crypto", "keccak256", crypto::keccak256),
+    let sui_natives: &[(&str, &str, NativeFunction)] = &[
+        ("crypto", "ecrecover", make_native!(crypto::ecrecover)),
```

Comment:
Why did we change the way that the `NativeFunctionTable` is constructed? Is this strictly necessary for the upstream Move version pull?

Reply:
Yes. It has some nice properties too
See https://github.com/move-language/move/pull/281#issue-1309141729